pkg/controller/node/node_controller.go

```diff
@@ -1079,8 +1214,8 @@ func getAllCandidateMachines(pool *mcfgv1.MachineConfigPool, nodesInPool []*core
 }
 
 // getCandidateMachines returns the maximum subset of nodes which can be updated to the target config given availability constraints.
-func getCandidateMachines(pool *mcfgv1.MachineConfigPool, nodesInPool []*corev1.Node, maxUnavailable int) []*corev1.Node {
-	nodes, capacity := getAllCandidateMachines(pool, nodesInPool, maxUnavailable)
+func getCandidateMachines(pool *mcfgv1.MachineConfigPool, config *mcfgv1alpha1.MachineOSConfig, build *mcfgv1alpha1.MachineOSBuild, nodesInPool []*corev1.Node, maxUnavailable int, layered bool) []*corev1.Node {
```

Comment:
suggestion (non-blocking): Make `getCandidateMachines()` accept a struct like `candidateMachineOpts` or similar. This makes it very clear at each call-site what each parameter is for.

(This does not need to be done right now though.)